common/api/presentation-components.api.md

```diff
@@ -514,7 +522,7 @@ export interface UnifiedSelectionTreeEventHandlerParams {
 // @beta
 export function useControlledTreeFiltering(props: ControlledTreeFilteringProps): {
     nodeHighlightingProps: HighlightableTreeProps | undefined;
-    filteredNodeLoader: AbstractTreeNodeLoaderWithProvider<IPresentationTreeDataProvider> | PagedTreeNodeLoader<FilteredPresentationTreeDataProvider>;
+    filteredNodeLoader: AbstractTreeNodeLoaderWithProvider<IPresentationTreeDataProvider>;

```

Comment:
No because if filter is not applied it will return `AbstractTreeNodeLoaderWithProvider<IPresentationTreeDataProvider>`.

If this type is wrong then I think it should be `AbstractTreeNodeLoaderWithProvider<IPresentationTreeDataProvider> | AbstractTreeNodeLoaderWithProvider<IFilteredPresentationTreeDataProvider>`. However it was inferred as `AbstractTreeNodeLoaderWithProvider<IPresentationTreeDataProvider>` I believe because `IFilteredPresentationTreeDataProvider` can be assigned to `IPresentationTreeDataProvider` so the most abstract type was chosen. 

I can change it to `AbstractTreeNodeLoaderWithProvider<IPresentationTreeDataProvider> | AbstractTreeNodeLoaderWithProvider<IFilteredPresentationTreeDataProvider>` if you think it would help consumers. I think in either case check or type guard is needed to determine which provider was returned.

Reply:
Thanks. I think it's fine - I should've read the docs of `useControlledTreeFiltering` which explains this.